app/com/arpnetworking/rollups/RollupExecutor.java

```diff
@@ -156,9 +158,49 @@ private void fetchRollup() {
     private void handleFinishRollup(final FinishRollupMessage message) {
         _metrics.recordCounter("rollup/executor/finish_rollup_message/received", 1);
         _rollupManager.tell(message, getSelf());
+        final RollupDefinition defn = message.getRollupDefinition();
+        if (shouldRequestConsistencyCheck(message)) {
```

Comment:
This seems like it should be the responsibility of the RollupManager. Then you don't have a check for rollup failure in two places.